k8s/k8s.go

```diff
@@ -250,6 +263,118 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 	return result, nil
 }
 
+// ipAddrsResult parses the ipAddrs annotation and calls IPAM plugin for
+// each IP passed to it by setting CNI_ARGS IP=ip_addr, and returns the type.Result.
+// Example annotation value string: "[\"10.0.0.1\", \"2001:db8::1\"]"
+func ipAddrsResult(ipAddrs string, conf utils.NetConf, args *skel.CmdArgs, logger *log.Entry) (*types.Result, error) {
+	var ips []string
+
+	err := json.Unmarshal([]byte(ipAddrs), &ips)
+	if err != nil {
+		logger.WithField("annotation", ipAddrs).Error("Failed to parse ipAddrs as json")
+		return nil, fmt.Errorf("Failed to parse ipAddrs: %s", err)
+	}
+
+	result := types.Result{}
+
+	// annotation value can't be empty.
+	if len(ips) == 0 {
+		logger.WithField("annotation", "cni.projectcalico.org/ipAddrs").Error("No IPs specified")
+		return nil, fmt.Errorf("Annotation \"cni.projectcalico.org/ipAddrs\" specified but empty")
+	}
+
+	log.Debugf("IPs passed through ipAddrs annotation: %s", ipAddrs)
+
+	// Go through all the IPs passed in as annotation value and call IPAM plugin
+	// for each, and populate the result variable with IP4 and/or IP6 IPs returned
+	// from the IPAM plugin. We also make sure there is only one IPv4 and/or one IPv6
+	// passed in, since CNI spec only supports one of each right now.
+	for _, ip := range ips {
+		ipAddr := net.ParseIP(ip)
+		if ipAddr == nil {
+			logger.WithField("IP", ip).Error("Invalid IP format")
+			return nil, fmt.Errorf("Invalid IP format: %s", ip)
```

Comment:
`net.ParseIP` doesn't return an error. It just sets the result to `nil` if it's not valid